packages/core/src/lib/TwaManifest.ts

```diff
@@ -315,6 +315,139 @@ export class TwaManifest {
     const json = JSON.parse((await fs.promises.readFile(fileName)).toString());
     return new TwaManifest(json);
   }
+
+  /**
+   * Merges the Twa Manifest with the web manifest. Ignores the specified fields.
+   *
+   * @param {string[]} fieldsToIgnore the fields which needs to be ignored.
+   * @param {URL} webManifestUrl the URL where the webmanifest is available.
+   * @param {WebManifest} webManifest the Web Manifest, used as a base for the update of
+   *    the TWA Manifest.
+   * @param {TwaManifest} oldTwaManifest current Twa Manifest.
+   */
+  static merge(fieldsToIgnore: string[], webManifestUrl: URL
+      , webManifest: WebManifestJson, oldTwaManifest: TwaManifest): void {
+    let shortcuts: ShortcutInfo[] = [];
+    let name;
+    let launcherName;
+    let display;
+    let themeColor;
+    let backgroundColor;
+    let fullStartUrl: URL;
+    let startUrl;
+    let icon: WebManifestIcon | null = null;
+    let maskableIcon: WebManifestIcon | null = null;
+    let monochromeIcon: WebManifestIcon | null = null;
+
+    const packageId = oldTwaManifest.packageId;
+    const host = oldTwaManifest.host;
+    const navigationColor = oldTwaManifest.navigationColor.hex();
+    const navigationColorDark = oldTwaManifest.navigationColorDark.hex();
+    const navigationDividerColor = oldTwaManifest.navigationDividerColor.hex();
+    const navigationDividerColorDark = oldTwaManifest.navigationDividerColorDark.hex();
+    const appVersionName = oldTwaManifest.appVersionName;
+    const signingKey = oldTwaManifest.signingKey;
+    const splashScreenFadeOutDuration = oldTwaManifest.splashScreenFadeOutDuration;
+    const enableNotifications = oldTwaManifest.enableNotifications;
+    const twaManifestShortcuts = shortcuts;
+    const twaManifestWebManifestUrl = oldTwaManifest.webManifestUrl!.toString();
+
+    function resolveIconUrl(icon: WebManifestIcon | null): string | undefined {
+      return icon ? new URL(icon.src, webManifestUrl).toString() : undefined;
+    }
+
+    if (!('shortcuts' in fieldsToIgnore)) {
+      for (let i = 0; i < (webManifest.shortcuts || []).length; i++) {
+        const s = webManifest.shortcuts![i];
+        try {
+          const shortcutInfo = ShortcutInfo.fromShortcutJson(webManifestUrl, s);
+          if (shortcutInfo != null) {
+            shortcuts.push(shortcutInfo);
+          }
+        } catch (err) {
+          TwaManifest.log.warn(`Skipping shortcut[${i}] for ${err.message}.`);
+        }
+        if (shortcuts.length === 4) {
+          break;
+        }
+      }
+    } else {
+      shortcuts = oldTwaManifest.shortcuts;
+    }
+
+    if (!('name' in fieldsToIgnore)) {
+      name = webManifest['name'] || webManifest['short_name'] || DEFAULT_APP_NAME;
```

Comment:
In all of these blocks, should it not be:

```
name = webManifest['name'] || webManifest['short_name'] || oldTwaManifest.name;
```

We want new values to override old ones, but I don't think we should ever have a default value override an old value.